lib/beaker-answers/versions/version20162.rb

```diff
@@ -26,7 +28,55 @@ def generate_answers
+        hiera_hash = {}
+        hiera_hash["console_admin_password"] = answer_for(@options, "console_admin_password")
+        hiera_hash["puppet_enterprise::use_application_services"] = answer_for(@options, "puppet_enterprise::use_application_services")
+        hiera_hash["puppet_enterprise::certificate_authority_host"] = answer_for(@options, "puppet_enterprise::certificate_authority_host", master.name)
+        hiera_hash["puppet_enterprise::puppet_master_host"] = answer_for(@options, "puppet_enterprise::puppet_master_host", master.name)
+        hiera_hash["puppet_enterprise::console_host"] = answer_for(@options, "puppet_enterprise::console_host", console.name)
+        hiera_hash["puppet_enterprise::puppetdb_host"] = answer_for(@options, "puppet_enterprise::puppetdb_host", puppetdb.name)
+        hiera_hash["puppet_enterprise::database_host"] = answer_for(@options, "puppet_enterprise::database_host", puppetdb.name)
+        hiera_hash["puppet_enterprise::pcp_broker_host"] = answer_for(@options, "puppet_enterprise::pcp_broker_host", master.name)
+        hiera_hash["puppet_enterprise::mcollective_middleware_hosts"] = [answer_for(@options, "puppet_enterprise::mcollective_middleware_hosts", master.name)]
+
+        # Database names/users. Required for password and cert-based auth
+        hiera_hash["puppet_enterprise::puppetdb_database_name"] = answer_for(@options, "puppet_enterprise::puppetdb_database_name")
+        hiera_hash["puppet_enterprise::puppetdb_database_user"] = answer_for(@options, "puppet_enterprise::puppetdb_database_user")
+        hiera_hash["puppet_enterprise::classifier_database_name"] = answer_for(@options, "puppet_enterprise::classifier_database_name")
+        hiera_hash["puppet_enterprise::classifier_database_user"] = answer_for(@options, "puppet_enterprise::classifier_database_user")
+        hiera_hash["puppet_enterprise::activity_database_name"] = answer_for(@options, "puppet_enterprise::activity_database_name")
+        hiera_hash["puppet_enterprise::activity_database_user"] = answer_for(@options, "puppet_enterprise::activity_database_user")
+        hiera_hash["puppet_enterprise::rbac_database_name"] = answer_for(@options, "puppet_enterprise::rbac_database_name")
+        hiera_hash["puppet_enterprise::rbac_database_user"] = answer_for(@options, "puppet_enterprise::rbac_database_user")
+        hiera_hash["puppet_enterprise::orchestrator_database_name"] = answer_for(@options, "puppet_enterprise::orchestrator_database_name")
+        hiera_hash["puppet_enterprise::orchestrator_database_user"] = answer_for(@options, "puppet_enterprise::orchestrator_database_user")
+
+        # We only need to specify passwords if we are using password auth
+        unless @options[:database_cert_auth]
+          hiera_hash["puppet_enterprise::puppetdb_database_password"] = answer_for(@options, "puppet_enterprise::puppetdb_database_password")
```

Comment:
The `DEFAULT_ANSWERS` hashes are only called if we use the `answer_for` method from `BeakerAnswers::Answers`.